src/com/sun/ts/tests/websocket/spec/servercontainer/upgradehttptowebsocket/WSTestServlet.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package com.sun.ts.tests.websocket.spec.servercontainer.upgradehttptowebsocket;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.websocket.DeploymentException;
+import jakarta.websocket.server.ServerContainer;
+import jakarta.websocket.server.ServerEndpointConfig;
+
+@WebServlet("/TCKTestServlet")
+public class WSTestServlet extends HttpServlet {
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    ServerContainer sc = (ServerContainer) getServletContext().getAttribute("jakarta.websocket.server.ServerContainer");
+    ServerEndpointConfig sec = ServerEndpointConfig.Builder.create(WSTestServer.class, "/TCKTestServer").build();
```

Comment:
@markt-asf The test application war is `ws_spec_upgradehttp_web.war`. So it creates context `ws_spec_upgradehttp_web` for the endpoints. The `TCKTestServer` endpoint gets deployed when the war is deployed. 
But then, `ServerEndpointConfig.Builder.create(WSTestServer.class, "/TCKTestServer").build()` creates the endpoint with the `/TCKTestServer` path and it is not matched with `ws_spec_upgradehttp_web/TCKTestServer`. Should the path here contain the context path?

Reply:
No. As per the Javadoc for Builder, the path is relative to the "web socket root" which is, as per section 6.5 of the WebSocket spec, the Servlet context root of the web application.